app/uk/gov/hmrc/customs/notification/connectors/InternalPushConnector.scala

```diff
@@ -64,7 +64,7 @@ class InternalPushConnector @Inject()(http: HttpClient,
     }
 
     unexpectedExceptionCatcher { () =>
-      http.POSTString[HttpResponse](pnr.body.url, pnr.body.xmlPayload)
+      http.POSTString[HttpResponse](pnr.body.url.toString, pnr.body.xmlPayload)
```

Comment:
One of the upsides of passing in a correctly-formed url to http-verbs is that it's no longer possible to get an exception thrown. I suggest that the `unexpectedExceptionCatcher` code is removed.